chat/src/commonMain/kotlin/com/ably/chat/Messages.kt

```diff
@@ -433,16 +433,20 @@ internal class DefaultMessages(
 
             val data = parsePubSubMessageData(eventType, pubSubMessage.data)
             val chatMessage = DefaultMessage(
-                createdAt = pubSubMessage.createdAt,
+                timestamp = pubSubMessage.timestamp,
                 clientId = pubSubMessage.clientId,
                 serial = pubSubMessage.serial,
                 text = data.text,
                 metadata = data.metadata ?: MessageMetadata(),
                 headers = pubSubMessage.extras?.asJsonObject()?.tryAsJsonValue()?.tryAsJsonObject()?.get("headers")?.toMap() ?: mapOf(),
                 action = pubSubMessage.action,
-                version = pubSubMessage.version,
-                timestamp = pubSubMessage.timestamp,
-                operation = pubSubMessage.operation,
+                version = DefaultMessageVersion(
+                    serial = pubSubMessage.version.serial ?: pubSubMessage.serial,
```

Comment:
For chat-js, we just follow the PubSub API and have this field as nullable (we don't implement filling in from `serial`)